From ae9708cbb35d5e444348e6238efe4094a70f6b69 Mon Sep 17 00:00:00 2001
From: "github-actions[bot]" <github-actions[bot]@users.noreply.github.com>
Date: Thu, 12 Feb 2026 19:10:50 -0700
Subject: [PATCH 1/2] cmake: include xproinc, call xpExternPackage()

---
 CMakeLists.txt | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16c3f01..083db72 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,6 @@
 # CMakeLists.txt
 cmake_minimum_required(VERSION 3.31)
+set(CMAKE_PROJECT_TOP_LEVEL_INCLUDES .devcontainer/cmake/xproinc.cmake)
 project(MyProject)
 
 # Find packages
@@ -13,10 +14,15 @@ target_link_libraries(app PRIVATE
     fmt::fmt
     spdlog::spdlog
     nlohmann_json::nlohmann_json)
-set_target_properties(app PROPERTIES DEBUG_POSTFIX -d)
+set_target_properties(app PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
 
-include(GNUInstallDirs)
-set(CMAKE_INSTALL_CMAKEDIR ${CMAKE_INSTALL_DATADIR}/cmake)
 set(targetsFile ${PROJECT_NAME}-targets)
+xpExternPackage(REPO_NAME tutorial NAMESPACE xp
+  TARGETS_FILE ${targetsFile} EXE app
+  BASE v1.0 XPDIFF "patch" PVT_DEPS fmt spdlog nlohmann_json
+  WEB "https://github.com/smanders/tutorial" UPSTREAM "github.com/externpro/tutorial"
+  DESC "A simple externpro tutorial project"
+  LICENSE "[MIT](https://github.com/externpro/tutorial?tab=MIT-1-ov-file#readme 'MIT License')"
+  )
 install(TARGETS app EXPORT ${targetsFile} DESTINATION ${CMAKE_INSTALL_BINDIR})
-install(EXPORT ${targetsFile} DESTINATION ${CMAKE_INSTALL_CMAKEDIR} NAMESPACE bar::)
+install(EXPORT ${targetsFile} DESTINATION ${CMAKE_INSTALL_CMAKEDIR} NAMESPACE xp::)
-- 
2.52.0

